ass2.cpp

```diff
@@ -0,0 +1,209 @@
+#include <iostream>
+#include <sstream>
+
+using namespace std;
+
+// TODO: Make two classes for each task with tests
+// TODO: Make comments for each test case
+
+// Task 1
+// 
+// Given a Binary Tree and a key, write a function that prints
+// all the ancestors of the key in the givenbinary tree.
+//
+
+struct node {
+    
+    int data;
+    struct node* left_child = nullptr;
+    struct node* right_child = nullptr;
+    
+    node(int val) {
+        data = val;
+    }
+};
+
+bool have_ancestors(struct node* root, int node_data) {
```

Comment:
The name of the function is somewhat misleading, since it prints the nodes, besides the return value...